### PR TITLE
Chore!: Migrate docker container registry

### PR DESCRIPTION
**Pull request by IsaacCalligeros95**
Migrates the docker container registry to the plugin framework.

### CODE REVIEW

File: docs/resources/docker_container_registry.md [hunk 1]
```diff
@@ -2,12 +2,12 @@
 page_title: "octopusdeploy_docker_container_registry Resource - terraform-provider-octopusdeploy"
 subcategory: "Feeds"
 description: |-
-  This resource manages a Docker Container Registry in Octopus Deploy.
+
```
Comment (benPearce1) [inline docs/resources/docker_container_registry.md:5]: looks like all the descriptions are being dropped from the docs

**Review by benPearce1 — APPROVED** [review]
tested locally. LGTM

File: octopusdeploy_framework/resource_docker_container_registry.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+func NewDockerContainerRegistryFeedResource() resource.Resource {
+	return &dockerContainerRegistryFeedTypeResource{}
+}
+
+func (r *dockerContainerRegistryFeedTypeResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = util.GetTypeName("docker_container_registry")
+}
+
+func (r *dockerContainerRegistryFeedTypeResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
```
Comment (hnrkndrssn) [inline octopusdeploy_framework/resource_docker_container_registry.go:30]: This needs a `Description` field with the description from the old schema file.

Comment (IsaacCalligeros95) [reply octopusdeploy_framework/resource_docker_container_registry.go:30]: I've fixed the descriptions across all the feeds and other resources I've migrated [here](https://github.com/OctopusDeployLabs/terraform-provider-octopusdeploy/pull/702/commits/124b893cd90260c9ba7612627cfe5b4296a63466).

**Pull request merged**